mullvad-relay-selector/src/relay_selector/helpers.rs

```diff
@@ -185,45 +185,29 @@ fn get_shadowsocks_obfuscator_inner(
 ///
 /// # Returns
 /// - A randomly selected port number within the given ranges.
-///
-/// # Panic
-/// - If port ranges contains no ports, this function panics.
-pub fn select_random_port(port_ranges: &[(u16, u16)]) -> Result<u16, Error> {
-    let get_port_amount = |range: &(u16, u16)| -> u64 { 1 + range.1 as u64 - range.0 as u64 };
-    let port_amount: u64 = port_ranges.iter().map(get_port_amount).sum();
-
-    if port_amount < 1 {
-        return Err(Error::PortSelectionAlgorithm);
-    }
-
-    let mut port_index = rand::thread_rng().gen_range(0..port_amount);
-
-    for range in port_ranges.iter() {
-        let ports_in_range = get_port_amount(range);
-        if port_index < ports_in_range {
-            return Ok(port_index as u16 + range.0);
-        }
-        port_index -= ports_in_range;
-    }
-    Err(Error::PortSelectionAlgorithm)
-}
-
-pub fn port_in_range(port: u16, port_ranges: &[(u16, u16)]) -> bool {
+/// - An error if `port_ranges` is empty.
+pub fn select_random_port(port_ranges: &[RangeInclusive<u16>]) -> Result<u16, Error> {
     port_ranges
         .iter()
-        .any(|range| (range.0 <= port && port <= range.1))
+        .flat_map(|range| range.clone())
```

Comment:
Elegant solution!

Reply:
Perhaps a small change to improve readability a bit
```rust
    port_ranges
        .iter()
        .cloned()
        .flatten() // Flatten to iter over all ports
        .choose(&mut rand::thread_rng())
        .ok_or(Error::NoMatchingPort)
```

Reply:
Nice. Fixed.